redis/client.go

```diff
@@ -216,7 +216,9 @@ func (c *Client) XReadUntilResult(ctx context.Context, a *redis.XReadArgs) ([]re
 		cmd := c.XRead(ctx, a)
 		streams, err := cmd.Result()
 		if err != nil {
-			if errors.Is(err, redis.Nil) {
+			// context.DeadlineExceeded implements both Temporary() and Timeout() which makes it retryable.
```

Comment:
I had to read through the PR and issue and all the comments to understand this change. Please adjust the comment accordingly:

- Why we need to explicitly retry here.
- Why the `ctx.Err() == nil` check is necessary.

Please also adjust the commit description with proper details for this change. The commit message is misleading as well.